controllers/azurestackedgemachine_controller.go

```diff
@@ -239,117 +239,117 @@ func (r *AzureMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, ret
-        conditions.MarkTrue(machineScope.AzureMachine, infrav1.VMRunningCondition)
+        conditions.MarkTrue(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition)
         machineScope.SetReady()
     case infrav1.VMStateCreating:
         machineScope.V(2).Info("VM is creating", "id", machineScope.GetVMID())
-        conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMNCreatingReason, clusterv1.ConditionSeverityInfo, "")
+        conditions.MarkFalse(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition, infrav1.VMNCreatingReason, clusterv1.ConditionSeverityInfo, "")
         machineScope.SetNotReady()
     case infrav1.VMStateUpdating:
         machineScope.V(2).Info("VM is updating", "id", machineScope.GetVMID())
-        conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMNUpdatingReason, clusterv1.ConditionSeverityInfo, "")
+        conditions.MarkFalse(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition, infrav1.VMNUpdatingReason, clusterv1.ConditionSeverityInfo, "")
         machineScope.SetNotReady()
     case infrav1.VMStateDeleting:
         machineScope.Info("Unexpected VM deletion", "id", machineScope.GetVMID())
-        r.Recorder.Eventf(machineScope.AzureMachine, corev1.EventTypeWarning, "UnexpectedVMDeletion", "Unexpected Azure VM deletion")
-        conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMDDeletingReason, clusterv1.ConditionSeverityWarning, "")
+        r.Recorder.Eventf(machineScope.AzureStackEdgeMachine, corev1.EventTypeWarning, "UnexpectedVMDeletion", "Unexpected Azure VM deletion")
+        conditions.MarkFalse(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition, infrav1.VMDDeletingReason, clusterv1.ConditionSeverityWarning, "")
         machineScope.SetNotReady()
     case infrav1.VMStateFailed:
         machineScope.Error(errors.New("Failed to create or update VM"), "VM is in failed state", "id", machineScope.GetVMID())
-        r.Recorder.Eventf(machineScope.AzureMachine, corev1.EventTypeWarning, "FailedVMState", "Azure VM is in failed state")
+        r.Recorder.Eventf(machineScope.AzureStackEdgeMachine, corev1.EventTypeWarning, "FailedVMState", "Azure VM is in failed state")
         machineScope.SetFailureReason(capierrors.UpdateMachineError)
         machineScope.SetFailureMessage(errors.Errorf("Azure VM state is %s", machineScope.VMState()))
-        conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMProvisionFailedReason, clusterv1.ConditionSeverityWarning, "")
+        conditions.MarkFalse(machineScope.AzureStackEdgeMachine, infrav1.VMRunningCondition, infrav1.VMProvisionFailedReason, clusterv1.ConditionSeverityWarning, "")
         machineScope.SetNotReady()
         // If VM failed provisioning, delete it so it can be recreated
         err := ams.DeleteVM(ctx)
```

Comment:
Should we also release IP back to the pool

Reply:
In this case, the VM and the disk will be deleted. But the network interface will not be deleted. The controller will retry creating the VM with the same network interface. Hence we dont release IP back to the pool. Do you think this is fine?

Reply:
or we need to change the logic here?